File: Assets/BeatTest.cs
using UnityEngine;
using System.Collections;

public class BeatTest : MonoBehaviour {

    public float t = 0.0f;
    public float deltaTime;
    public float bpm = 0f;
    public float beat = 0.0f;
    public float beatTot = 0.0f;
    public AudioSource Mo;
    private float expectedTime;
    private float timeOffset;
	// Use this for initialization
	void Start () {
        timeOffset = 60.0f / bpm;
        Debug.Log(timeOffset);
        expectedTime = timeOffset;
	}
	
	// Update is called once per frame
	void Update () {
	    t = Mo.time;

        if (t > expectedTime)
        {
            Debug.Log("Beat");
            expectedTime += timeOffset;
        }
	}
}

File: Assets/BeatTest.js
var beat = 0.0;
var beatSub = 0;
var beatTot = 0.0;
static var rhythmAble = false;
var guiPop : GUITexture;
var Mo : AudioSource;
 
function Start()
{
    GetComponent.<AudioSource>().Play();
}
 
function Update()
{
    beat = Mo.time;
    beatSub = Mo.time;
    beatTot = beat - beatSub;
    guiPop.pixelInset.x = beatTot*100 + 220;
    if ((beatTot > 0) && (beatTot < .4)) {
        rhythmAble = true;
    } else {
        rhythmAble = false;
    }
    var beatStr = beatTot.ToString();
    GetComponent.<GUIText>().text = beatStr;
}   